mix/CodeModel.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+    This file is part of cpp-ethereum.
+
+    cpp-ethereum is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    cpp-ethereum is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file CodeModel.cpp
+ * @author Arkadiy Paronyan arkadiy@ethdev.com
+ * @date 2014
+ * Ethereum IDE client.
+ */
+
+#include <sstream>
+#include <QDebug>
+#include <QApplication>
+#include <QtQml>
+#include <libsolidity/CompilerStack.h>
+#include <libsolidity/SourceReferenceFormatter.h>
+#include <libevmcore/Instruction.h>
+#include "QContractDefinition.h"
+#include "QFunctionDefinition.h"
+#include "QVariableDeclaration.h"
+#include "CodeHighlighter.h"
+#include "CodeModel.h"
+
+using namespace dev::mix;
+
+void BackgroundWorker::queueCodeChange(int _jobId, QString const& _content)
+{
+	m_model->runCompilationJob(_jobId, _content);
+}
+
+CompilationResult::CompilationResult():
+	QObject(nullptr), m_successfull(false),
+	m_codeHash(qHash(QString())),
+	m_contract(new QContractDefinition()),
+	m_codeHighlighter(new CodeHighlighter())
+{}
+
+CompilationResult::CompilationResult(const solidity::CompilerStack& _compiler):
+	QObject(nullptr), m_successfull(true), m_codeHash(qHash(QString()))
+{
+	if (!_compiler.getContractNames().empty())
+	{
+		m_contract.reset(new QContractDefinition(&_compiler.getContractDefinition(std::string())));
+		m_bytes = _compiler.getBytecode();
+		m_assemblyCode = QString::fromStdString(dev::eth::disassemble(m_bytes));
+	}
+	else
+		m_contract.reset(new QContractDefinition());
+}
+
+CompilationResult::CompilationResult(CompilationResult const& _prev, QString const& _compilerMessage):
+	QObject(nullptr), m_successfull(false), m_codeHash(qHash(QString())),
```

Comment:
`successful` (one 'l')
stick to one-per-line or all-on-one-line. but don't mix the two.